src/main/java/duke/Event.java

```diff
@@ -0,0 +1,36 @@
+package duke;
+
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+
+/**
+ * A type of Task that will happen at some point in the future.
+ */
+public class Event extends Task {
+    protected String at;
+    protected LocalDate dateOfEvent;
+    public Event(String description,String at) {
+        super(description, "E");
+
+        String dateString = Parser.extractDate(at);
+        if (!dateString.equals("")) {
+            this.dateOfEvent = Parser.parseDate(dateString);
+            this.at = at;
+        } else {
+            throw new IllegalArgumentException("Sorry Unable to Parse Date for Event. Did you put in yyyy-mm-dd format?");
```

Comment:
Please use 2 lines instead for long statements like you have done in the last test in DeadLineTest.java